lib/cookbook-omnifetch/artifactserver.rb

```diff
@@ -39,14 +39,21 @@ def installed?
     # @return [void]
     def install
       FileUtils.mkdir_p(cache_root) unless cache_root.exist?
+      FileUtils.mkdir_p(staging_root) unless staging_root.exist?
+
+      Dir.mktmpdir(nil, staging_root) do |staging_dir|
+        http = http_client(uri)
+        http.streaming_request(nil) do |tempfile|
+          tempfile.close
+          # TODO: What are the ACLs supposed to be?  Windows might or might not
+          # recalculate your ACLs based on how the move was accomplished.
+          staging_tgz_path = File.join(staging_dir, "#{cache_key}.tgz")
+          # Might be atomic?
+          FileUtils.mv(tempfile.path, staging_tgz_path)
+          # Probably is atomic.
+          FileUtils.mv(staging_tgz_path, cache_path)
```

Comment:
Looks like this works around the Windows cross-device move in two ways when we only need one. Since the bug occurs when we:

1. Move a directory (as opposed to an individual file)
2. AND move the filesystem entry across devices

Then we only need to ensure that we either:

1. Only move a single file (by moving the tarball)
2. OR, explode the tarball in the default temp dir, but do a `cp_r` of the directory followed by `rm_rf` of the staged directory to a location where we can ensure atomic behavior of a `mv` to the final destination.